keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/DirichletVertexTest.java

```diff
@@ -0,0 +1,237 @@
+package io.improbable.keanu.vertices.dbl.probabilistic;
+
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.dbl.Nd4jDoubleTensor;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import org.apache.commons.math3.util.Pair;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.improbable.keanu.vertices.dbl.probabilistic.ProbabilisticDoubleTensorContract.sampleMethodMatchesLogProbMethodMultiVariate;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.number.IsCloseTo.closeTo;
+import static org.junit.Assert.assertEquals;
+
+public class DirichletVertexTest {
+
+    private KeanuRandom random;
+
+    @Before
+    public void setup() {
+        random = new KeanuRandom(1);
+    }
+
+    @Test
+    public void twoDimensionalDirichletEqualsABeta() {
+        double alpha = 0.4;
+        double beta = 1.;
+        BetaVertex betaVertex = new BetaVertex(alpha, beta);
+        DirichletVertex dirichletVertex = new DirichletVertex(alpha, beta);
+
+        Assert.assertEquals(betaVertex.logPdf(0.5), dirichletVertex.logPdf(DoubleTensor.create(new double[]{0.5, 0.5})), 1e-6);
+        Assert.assertEquals(betaVertex.logPdf(0.75), dirichletVertex.logPdf(DoubleTensor.create(new double[]{0.75, 0.25})), 1e-6);
+    }
+
+    @Test
+    public void flatUniformIfAllConcentrationValuesAreOne() {
+        DirichletVertex dirichlet = new DirichletVertex(1, 1);
+
+        double twoDimDirichletPdf1 = dirichlet.logPdf(DoubleTensor.create(new double[]{0.7, 0.3}));
+        double twoDimDirichletPdf2 = dirichlet.logPdf(DoubleTensor.create(new double[]{0.3, 0.7}));
+        double twoDimDirichletPdf3 = dirichlet.logPdf(DoubleTensor.create(new double[]{0.5, 0.5}));
+
+        Assert.assertTrue(twoDimDirichletPdf1 == twoDimDirichletPdf2 && twoDimDirichletPdf2 == twoDimDirichletPdf3);
+
+        dirichlet = new DirichletVertex(new int[]{1, 4}, 1);
+
+        double fourDimDirichletPdf1 = dirichlet.logPdf(DoubleTensor.create(new double[]{0.1, 0.2, 0.3, 0.4}));
+        double fourDimDirichletPdf2 = dirichlet.logPdf(DoubleTensor.create(new double[]{0.7, 0.1, 0.1, 0.1}));
+        double fourDimDirichletPdf3 = dirichlet.logPdf(DoubleTensor.create(new double[]{0.25, 0.25, 0.25, 0.25}));
+
+        Assert.assertTrue(fourDimDirichletPdf1 == fourDimDirichletPdf2 && fourDimDirichletPdf2 == fourDimDirichletPdf3);
+    }
+
+    @Test
+    public void matchesScipyDirichletLogPdf() {
+        DirichletVertex dirichlet = new DirichletVertex(2, 2);
+        double scipyPdf = 0.8662499999999997;
```

Comment:
This feels potentially fragile, but I can't really see a better way of dealing with this short of committing a script to generate the correct number at runtime.  If we have a different implementation, perhaps we could check a few more values rather than just one though?

Reply:
Is the problem that we don't have a Java library to check against, like apache?